### DISCUSSION TITLE
Transport doesn't update correctly after saving the changes in edit form

### DISCUSSION THREAD

**Discussion by donacrisante** [neuefische/questions#711] [Web] [answered]
Hello everybody!
I have noticed that my means of transport does not update properly when I want to change it. For example, if I made a trip with diesel car and want to correct it with train, I get train diesel or bicycle diesel, if I select "bicycle" :D.
I tried to fix the error with an if statement, but it doesn't work and I have no idea how else to fix it. Can anyone help me with this?

<img width="336" alt="Bildschirm­foto 2023-08-30 um 14 37 46" src="https://github.com/neuefische/questions/assets/128093782/13a82164-03fd-4cb6-b42c-a84ab691ca8b">

Thank you in advance!

Here my PR:
https://github.com/donacrisante/Capstone-project/pull/13

**Comment by Kheirah** [comment, marked as answer]
We implemented a quick and dirty fix in person. Here's the gist of it:
```js
newEntry.result = handleCalculateCo2(
      transport,
      km,
      fuel
    );
newEntry.fuel = fuel === "Select a car" ? "" : fuel;
```
Plus, protection against the missing `fuel` value when you first select "car" as the transport but hadn't had the time to select the fuel.
```js
function handleCalculateCo2(transport, km, fuel) {
    if (!fuel) return 0;
    const selectedTransport = transport === "car" ? fuel : transport;
    const kmNumber = parseFloat(km.replace(",", "."));
    if (!isNaN(kmNumber)) {
      return calculator[selectedTransport](kmNumber);
    }
    return 0;
  }
```
